Rakefile

```
@@ -13,15 +13,17 @@ if RUBY_PLATFORM =~ /java/
   end
 
   task :build => "#{extask.lib_dir}/#{extask.name}.jar"
-else
+elsif RUBY_ENGINE == "ruby"
   require 'rake/extensiontask'
   extask = Rake::ExtensionTask.new(name) do |x|
     x.lib_dir << "/#{RUBY_VERSION}/#{x.platform}"
   end
+else
+  task :compile
 end
 Rake::TestTask.new(:test) do |t|
-  ENV["RUBYOPT"] = "-I" + [extask.lib_dir, "test/lib"].join(File::PATH_SEPARATOR)
-  t.libs << extask.lib_dir
+  ENV["RUBYOPT"] = "-I" + [*(extask.lib_dir if extask), "test/lib"].join(File::PATH_SEPARATOR)
+  t.libs << extask.lib_dir if extask
```

Could you simplify this for easy to read?

```suggestion
  if extask
    ENV["RUBYOPT"] = "-I" + [extask.lib_dir, "test/lib"].join(File::PATH_SEPARATOR)
    t.libs << extask.lib_dir
  else
    ENV["RUBYOPT"] = "-Itest/lib"
  end
```